### ISSUE TITLE
feat: wire real DRR computation into FastAPI backend (Phase 2)

### ISSUE THREAD

**Issue by roulbac** [Roulbac/2D3DAutoReg#3]
## Context

Phase 1 delivered a working React UI + FastAPI stub that returns a fixed placeholder PNG. This issue tracks Phase 2: replacing the stub with real cone-beam DRR computation driven by the `HN_P001.nii.gz` CT volume.

The legacy desktop app used **Numba JIT + pycuda** for ray tracing. This plan replaces that with a vectorized **PyTorch** implementation that auto-selects `cuda → mps (Apple Silicon) → cpu`, keeping it portable across all hardware.

No frontend changes needed — the API contract is unchanged.

---

## Files to Change

| File | Action |
|------|--------|
| `backend/app/drr_engine.py` | **Create** — PyTorch DRR renderer |
| `backend/app/main.py` | **Modify** — add lifespan, wire engine, drop stub |
| `backend/requirements.txt` | **Modify** — add `torch` (CPU), `SimpleITK` |
| `backend/Dockerfile` | **Modify** — split pip for torch CPU index, add `libgomp1` |
| `docker-compose.yml` | **Modify** — mount `Test_Data/`, add `NIFTI_PATH` env var |

---

### Pipeline
1. **Load volume** via SimpleITK (same as legacy `read_rho`): `(Z,Y,X) → (X,Y,Z)` float32, stored as `(1,1,Nx,Ny,Nz)` tensor on device
2. **Auto AP camera** — no calibration file needed; intrinsics/extrinsics computed from volume physical extent + SID=1000mm
3. **Pose application** — `R = Rz·Ry·Rx` centered on volume centroid (matching frontend convention)
4. **Vectorized ray generation** — pixel grid unprojected through `K⁻¹`, rotated to world space, no Python pixel loop
5. **AABB slab intersection** — fully vectorized; produces `t_near`, `t_far`, `valid` masks
6. **Volume sampling** — `torch.nn.functional.grid_sample` with trilinear interpolation (replaces Numba DDA loop)
7. **Beer–Lambert accumulation** — same constants as legacy `raybox.py` (μ_water=0.037, HU threshold=500)
8. **Output** — normalize to [0,1], encode as grayscale PNG → base64 data URI

### Key Implementation Notes
- `grid_sample` 5D axis ordering: volume stored as `(N,C,D=Nx,H=Ny,W=Nz)`, so grid coords must be `(norm_z, norm_y, norm_x)` — reversed from world XYZ
- Single camera: skip `CameraSet.make_p_pprime` (requires two cameras); implement pose math directly
- Memory: at 512×512 px × 256 samples, sample grid ≈ 268 MB — tile in 64×64 blocks for CPU if needed
- Startup latency: volume load + device transfer ~5–15s; FastAPI `lifespan` blocks requests until ready (correct)

---

## Dependency Changes

**`requirements.txt`**
```
torch==2.3.1+cpu   # ~700 MB installed vs 2.5 GB for CUDA
SimpleITK==2.3.1
```

**`Dockerfile`**
```dockerfile
RUN apt-get install -y libgomp1          # required by PyTorch OpenMP
RUN pip install torch==2.3.1+cpu \
    --index-url https://download.pytorch.org/whl/cpu
RUN pip install -r requirements.txt
```

**`docker-compose.yml`**
```yaml
volumes:
  - ./Test_Data:/data:ro
environment:
  - NIFTI_PATH=/data/HN_P001.nii.gz
```

---

## Verification

```bash
# 1. Smoke test (local)
python -c "
from backend.app.drr_engine import DRREngine
e = DRREngine('Test_Data/HN_P001.nii.gz')
img = e.render(0,0,0,0,0,0)
assert img.shape == (512,512) and 0 <= img.min() and img.max() <= 1
print('PASS')
"

# 2. Pose sensitivity
# render(0,0,0,...) vs render(10,0,0,...) should differ by > 0.001 mean px

# 3. API round-trip
curl -s -X POST http://localhost:8000/api/drr/generate \
  -H 'Content-Type: application/json' \
  -d '{"pose":{"tx":0,"ty":0,"tz":0,"rx":0,"ry":0,"rz":0}}'

# 4. Visual sanity in UI
# Open http://localhost:3000 → Generate DRR
# Expect head/neck silhouette with bone contrast
# Slide tx → image shifts horizontally
# Slide rx → image shifts vertically
```
